examples/protonect/src/rgb_packet_stream_parser.cpp

```diff
@@ -58,49 +73,66 @@ void RgbPacketStreamParser::setPacketProcessor(BaseRgbPacketProcessor *processor
 
 void RgbPacketStreamParser::onDataReceived(unsigned char* buffer, size_t length)
 {
+  // if no data return
+  if (length == 0)
+    return;
+
   Buffer &fb = buffer_.front();
 
-  // package containing data
-  if(length > 0)
+  // iterate through each byte to detect footer
+  for (size_t i = 0; i < length; i++)
   {
-    if(fb.length + length <= fb.capacity)
-    {
-      memcpy(fb.data + fb.length, buffer, length);
+    RgbPacketFooter* footer = reinterpret_cast<RgbPacketFooter *>(&buffer[i]);
+
+    // if magic markers match
+    if (length - i >= sizeof(RgbPacketFooter) && footer->magic_header == 0x39393939 && footer->magic_footer == 0x42424242)
+    {  
+      memcpy(&fb.data[fb.length], buffer, length);
```

Comment:
This copies all remaining data to the fb. I guess it would be more correct to just copy i+sizeof(footer) and copy any remaining data to the new fb after the swap?